docs/proposals/backup/backup.md

```diff
@@ -0,0 +1,735 @@
+	// +optional
+	Phase string `json:"phase,omitempty"`
+
+	// Details provides a detailed status for each backup in each cluster.
+	// +optional
+	Details []*BackupDetails `json:"backupDetails,omitempty"`
+}
+
+type BackupDetails struct {
+	// ClusterName is the Name of the cluster where the backup is being performed.
+	// +optional
+	ClusterName string `json:"clusterName,omitempty"`
+
+	// ClusterKind is the kind of ClusterName recorded in Kurator.
+	// +optional
+	ClusterKind string `json:"clusterKind,omitempty"`
+
+	// BackupNameInCluster is the name of the backup being performed within this cluster.
+	// This BackupNameInCluster is unique in Storage.
+	// +optional
+	BackupNameInCluster string `json:"backupNameInCluster,omitempty"`
+
+	// BackupStatusInCluster is the current status of the backup performed within this cluster.
+	// +optional
+	BackupStatusInCluster *velerov1.BackupStatus `json:"backupStatusInCluster,omitempty"`
+}
+
+// BackupList contains a list of Backup.
+// +kubebuilder:object:root=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type BackupList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Backup `json:"items"`
+}
+```
+
+Here is the details about `Destination` and `ResourceFilter`
+
+```console
+
+// Destination defines a fleet or specific clusters.
+type Destination struct {
+	// Fleet is the name of fleet.
+	// The field, in combination with ClusterSelector, can determine a set of clusters.
+	// In addition to this approach, users can also directly specify clusters through the field Clusters.
+	// +optional
+	Fleet string `json:"fleet,omitempty"`
+
+	// ClusterSelector specifies the selectors to select the clusters within the fleet.
+	// If unspecified, all clusters in the fleet will be selected.
+	// The field will only take effect when Fleet is set.
+	// +optional
+	ClusterSelector *ClusterSelector `json:"clusterSelector,omitempty"`
+
+	// Clusters determine a set of clusters as destination clusters.
+	// The field will only take effect when Fleet is not set.
+	// +optional
+	Clusters []*corev1.ObjectReference `json:"clusters,omitempty"`
+}
+
+type ClusterSelector struct {
+	// MatchLabels is a map of {key,value} pairs. A single {key,value} in the matchLabels map is equivalent to an element of matchExpressions, whose key field is "key", the operator is "In", and the values array contains only "value".
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#label-selectors
+	// +optional
+	MatchLabels map[string]string `json:"matchLabels,omitempty"`
+}
+
+// Note: partly copied from https://github.com/"github.com/vmware-tanzu/velero/pkg/apis/backup_types.go
+type ResourceFilter struct {
+	// IncludedNamespaces is a slice of namespace names to include objects from.
+	// If empty, all namespaces are included.
+	// +optional
+	// +nullable
```

Comment:
what does nullable do

Reply:
this describe ` +nullable` come from velero.
the "+nullable" directive, when used in conjunction with "+optional", emphasizes that the field is not only optional but also can be set to a null value.

Reply:
Not necessary, it is not a formal comment like optional.